esp/services/ws_decoupledlogging/ws_decoupledloggingservice.cpp

```diff
@@ -294,6 +187,96 @@ bool CWSDecoupledLogEx::onPauseLog(IEspContext& context, IEspPauseLogRequest& re
     return true;
 }
 
+void CWSDecoupledLogEx::getSettingsForLoggingAgentsInGroup(WSDecoupledLogAgentGroup* group, StringArray* agentNames,
+    IArrayOf<IEspLogAgentGroupSetting>& groupSettingResp)
+{
+    IArrayOf<IEspLogAgentSetting> agentSettingResp;
+    if (!agentNames || !agentNames->ordinality())
+        getSettingsForAllLoggingAgentsInGroup(group, agentSettingResp);
```

Comment:
The original comment that led to this method (and pauseLoggingAgentsInGroup) also mentioned the handling of agents within a group. Just as this method is independent of how the group was chosen, the expectation was for a comparable method acting on an agent regardless of how it is chosen.